metaflow/plugins/kfp/kfp.py

```diff
@@ -366,7 +275,86 @@ def to_k8s_resource_format(resource: str, value: Union[int, float, str]) -> str:
 
         return resource_requirements
 
-    def create_kfp_components_from_graph(self) -> Dict[str, KfpComponent]:
+    def _create_flow_variables(self) -> FlowVariables:
+        flow_variables = FlowVariables(
+            flow_name=self.flow.name,
+            environment=self.environment.TYPE,
+            event_logger=self.event_logger.logger_type,
+            monitor=self.monitor.monitor_type,
+            namespace=self.namespace,
+            tags=list(self.tags),
+            package_commands=self._get_package_commands(
+                code_package_url=self.code_package_url,
+                environment=self.environment,
+            ),
+        )
+        return flow_variables
+
+    def _get_package_commands(
+        self,
+        code_package_url: str,
+        environment: MetaflowEnvironment,
+    ) -> str:
+        if self.s3_code_package:
+            cmd: List[str] = [
+                BASH_MFLOG_KFP,
+                "mkdir -p /opt/metaflow_volume/metaflow_logs",
+                "export MFLOG_STDOUT=/opt/metaflow_volume/metaflow_logs/mflog_stdout",
+            ]
+            cmd.extend(environment.get_package_commands(code_package_url))
+            return " && ".join(cmd)
+        else:
+            return " cd " + str(Path(inspect.getabsfile(self.flow.__class__)).parent)
+
+    def _create_step_variables(self, node: DAGNode) -> StepVariables:
+        """
+        Returns the Metaflow Node StepVariables, which is
+        used to run Metaflow on KFP "kfp_metaflow_step()"
+        """
+
+        task_id: str = graph_to_task_ids(self.graph)[node.name]
+        user_code_retries, total_retries = KubeflowPipelines._get_retries(node)
+        resource_requirements: Dict[str, str] = self._get_resource_requirements(node)
+
+        is_split_index: bool = (
+            True
+            if any(self.graph[n].type == "foreach" for n in node.in_funcs)
+            else False
+        )
+
+        return StepVariables(
+            clean_volume_cmd=KubeflowPipelines._get_clean_volume_cmd(
+                resource_requirements
+            ),
+            step_name=node.name,
+            is_split_index=is_split_index,
+            task_id=task_id,
+            task_id_template=self._get_task_id_template(node.name, task_id),
+            total_retries=total_retries,
+            user_code_retries=user_code_retries,
+        )
+
+    def _get_task_id_template(self, step_name: str, task_id: str) -> str:
+        if self.graph[step_name].is_inside_foreach:
+            return KfpForEachSplits.get_step_task_id(
+                task_id=task_id,
+                passed_in_split_indexes="{passed_in_split_indexes}",
+            )
+        else:
+            return task_id
```

Comment:
just as `_get_clean_volume_cmd` below, this doesn't need to be passed into `kfp_metaflow_step ` but can be moved to back to `def _command()` as `kfp_metaflow_step` will have access to the true `passed_in_split_indexes` and `task_id`.

Basically, we can delete `task_id_template` from `StepVariables`